packages/owlbot-bootstrapper/common-container/api-field-fetcher.ts

```diff
@@ -0,0 +1,197 @@
+// Copyright 2022 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import {Octokit} from '@octokit/rest';
+import {Storage} from '@google-cloud/storage';
+import {ApiFields, DriftApi, GHDir, GHFile} from './interfaces';
+import yaml from 'js-yaml';
+import {getWellKnownFileContents, INTER_CONTAINER_VARS_FILE} from './utils';
+import * as fs from 'fs';
+/**
+ * Fetches API-related information
+ *
+ * @param apiId the unique API ID
+ * @param storageClient a new instance of Storage
+ * @param octokit a new instance of Octokit
+ */
+export class ApiFieldFetcher {
+  apiId: string;
+  storageClient: Storage;
+  octokit: Octokit;
+
+  constructor(apiId: string, octokit: Octokit, storageClient: Storage) {
+    this.apiId = apiId;
+    this.octokit = octokit;
+    this.storageClient = storageClient;
+  }
+
+  /**
+   * Function that gets api information from a public DRIFT bucket, and saves it to a well-known location on disk
+   * @param apiId the unique API ID
+   * @param storageClient an instance of Google Cloud Storage
+   * @returns DriftApi info
+   */
+  public async _getDriftMetadata(
+    apiId: string,
+    storageClient: Storage
+  ): Promise<DriftApi> {
+    const bucket = 'devrel-prod-settings';
+    const file = 'apis.json';
+    const contents = await storageClient.bucket(bucket)?.file(file)?.download();
+    if (!contents) {
+      throw new Error('apis.json downloaded from Cloud Storage was empty');
+    }
+
+    const apis = JSON.parse(contents.toString()).apis;
+    return this._extractApiInfoFromJson(apis, apiId);
+  }
+  /**
+   * Function that extracts information for a given API from apis.json
+   *
+   * @param apis the apis.json file downloaded from drift as a JSON object
+   * @param apiId the unique identifier of the API
+   * @returns DriftAPI the fields as described in Drift
+   */
+  public _extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
```

Comment:
Same on this (public method with `_` name).
```suggestion
  public extractApiInfoFromJson(apis: DriftApi[], apiId: string): DriftApi {
```